File: java/Hello.java
import java.io.*;
import java.nio.MappedByteBuffer;
import java.nio.charset.Charset;
import java.nio.channels.FileChannel;

import com.google.gson.Gson;
import com.google.gson.JsonObject;
import com.google.gson.JsonArray;
import com.google.gson.JsonParser;

import java.net.URL;
import java.net.URLConnection;

import org.json.JSONArray;
import org.json.JSONObject;
import io.iron.ironworker.client.helpers.WorkerHelper;

public class Hello {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) throws Exception {
        System.out.println("Running worker");
        WorkerHelper helper = WorkerHelper.fromArgs(new String[0]);
        BufferedReader in = new BufferedReader(
                new InputStreamReader(System.in));
        StringBuffer buff = new StringBuffer();

        // Get payload data with the IronWorker helper
        PayloadData payload = helper.getPayload(PayloadData.class);

        System.out.println("Hello " + payload.getName());

    }

    private static void writeFile(String path, String content) throws IOException {
        Writer out = new OutputStreamWriter(new FileOutputStream(path));
        try {
            out.write(content);
        } finally {
            out.close();
        }

    }
}

File: kotlin/Hello.kt
import java.io.*
import java.nio.MappedByteBuffer
import java.nio.charset.Charset
import java.nio.channels.FileChannel

import com.google.gson.Gson
import com.google.gson.JsonObject
import com.google.gson.JsonArray
import com.google.gson.JsonParser

import java.net.URL
import java.net.URLConnection

import org.json.JSONArray
import org.json.JSONObject
import io.iron.ironworker.client.helpers.WorkerHelper

object Hello {

    @Throws(Exception::class)
    @JvmStatic
    fun main(args: Array<String>) {
        println("Running worker")

        val helper = WorkerHelper.fromArgs(arrayOfNulls<String>(0))
        val `in` = BufferedReader(
                InputStreamReader(System.`in`))
        val buff = StringBuffer()

        // Get payload data with the IronWorker helper
        val payload = helper.getPayload(PayloadData::class.java)

        println("Hello " + payload.name)
    }

    @Throws(IOException::class)
    private fun writeFile(path: String, content: String) {
        val out = OutputStreamWriter(FileOutputStream(path))
        try {
            out.write(content)
        } finally {
            out.close()
        }

    }
}
